### PR TITLE
SUS-3120 Rename local user related pages

### PR DESCRIPTION
**Pull request by mszabo-wikia**
On user rename completion, trigger background tasks to move user pages and similar content to the new user name.

https://wikia-inc.atlassian.net/browse/SUS-3120

### CODE REVIEW

File: includes/wikia/tasks/Tasks/RenameUserPagesTask.class.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	 * @return int[] array of wiki IDs
+	 */
+	public static function getTargetCommunities( string $oldUserName ) {
+		global $wgExternalDatawareDB;
+
+		$dbr = wfGetDB( DB_SLAVE, [], $wgExternalDatawareDB );
+
+		return $dbr->selectFieldValues( 'pages', 'page_wikia_id', [
+			'page_namespace' => static::NAMESPACES,
+			'page_title' => $oldUserName
```
Comment (macbre) [inline includes/wikia/tasks/Tasks/RenameUserPagesTask.class.php:28]: Should we change it to `LIKE` condition and move user page sub-pages as well?

Comment (macbre) [reply includes/wikia/tasks/Tasks/RenameUserPagesTask.class.php:28]: Oh, wait, it's covered in `renameLocalPages`

Comment (mszabo-wikia) [reply includes/wikia/tasks/Tasks/RenameUserPagesTask.class.php:28]: There shouldn't be a need to do that when checking for where to run task:
* if Wall root page does not exist, Wall is broken
* if blog listing page does not exist, Blogs are also broken
* if user does not have user page, user talk page, Message Wall, or blog, but somehow has user or user talk subpages... - do we want to check for such situation?

File: includes/wikia/tests/core/WikiaBaseTest.class.php [hunk 2]
```diff
@@ -453,26 +427,11 @@ public function __retrieveMessageMock( $key ) {
...
-
-		return $wgMemcMock;
-	}
-
 	/**
 	 * Mocks global $wgMemc->get() so it always returns null
 	 */
 	protected function disableMemCache() {
-		$this->mockGlobalVariable( 'wgMemc', $this->getMemCacheMock() );
+		$this->mockGlobalVariable( 'wgMemc', new EmptyBagOStuff() );
```
Comment (macbre) [inline includes/wikia/tests/core/WikiaBaseTest.class.php:434]: :+1:

File: extensions/wikia/UserRenameTool/tests/fixtures/dataware-pages.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  `page_namespace` int unsigned NOT NULL DEFAULT '0',
+  `page_title` varchar(255) NOT NULL,
+  `page_is_content` tinyint unsigned NOT NULL DEFAULT '0',
+  `page_is_redirect` tinyint unsigned NOT NULL DEFAULT '0',
+  `page_latest` int unsigned NOT NULL DEFAULT '0',
+  `page_last_edited` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (`page_wikia_id`,`page_id`)
+);
+
+CREATE INDEX `page_title_namespace_latest_idx` ON `pages` (`page_title`,`page_namespace`,`page_latest`);
```
Comment (macbre) [inline extensions/wikia/UserRenameTool/tests/fixtures/dataware-pages.sql:13]: :+1:

**Review by macbre — APPROVED** [review]

**Pull request merged**